src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php

```diff
@@ -303,6 +303,11 @@ public function get_items( $request ) {
 
 		if ( ! empty( $prepared_args['search'] ) ) {
 			$prepared_args['search'] = '*' . $prepared_args['search'] . '*';
+			/**
+			 * Specifying search fields directly, excluding user_email field for unlogged users, to avoid unwanted data exposure for public endpoint.
+			 * @link https://core.trac.wordpress.org/ticket/53784
+			 */
+			$prepared_args['search_columns'] = array( 'ID', 'user_login', 'user_url', 'user_nicename', 'display_name' );
```

Comment:
This should receive the same treatment as comments of allowing searching across all columns if the user has `list_users`.

Reply:
The `user_login` field should also be omitted from this list as it is only exposed in the `edit` context.

Reply:
I added additional fields for permissioned users in 530fcd3b8ab04f2c3807e580284f2891f93aefb7